src/services/GenerateVerifiableCredential.ts

```diff
@@ -0,0 +1,457 @@
+import { YOTI_CHECKS, YotiSessionDocument } from "../utils/YotiPayloadEnums";
+import {
+	VerifiedCredentialEvidence,
+	VerifiedCredentialSubject,
+} from "../utils/IVeriCredential";
+
+export class GenerateVerifiableCredential {
+  readonly logger: Logger;
+
+  private static instance: GenerateVerifiableCredential;
+
+  constructor(logger: Logger) {
+  	this.logger = logger;
+  }
+
+  static getInstance(logger: Logger): GenerateVerifiableCredential {
+  	if (!GenerateVerifiableCredential.instance) {
+  		GenerateVerifiableCredential.instance = new GenerateVerifiableCredential(
+  			logger,
+  		);
+  	}
+  	return GenerateVerifiableCredential.instance;
+  }
+
+	/**
+		Checks if Document contains a chip and if that chip is valid but checking the chip_csca_trusted field
+	*/
+  private doesDocumentContainValidChip(documentType: string, documentAuthenticityCheckBreakdown: { sub_check: string; result: string; }[]): boolean {
+  	const validDocumentTypes = ["PASSPORT", "NATIONAL_ID", "RESIDENCE_PERMIT"];
+	
+  	if (validDocumentTypes.includes(documentType)) {
+  		const validChip = documentAuthenticityCheckBreakdown.some(
+  			(subCheck: { sub_check: string; result: string }) =>
+  				subCheck.sub_check === YotiSessionDocument.CHIP_CSCA_TRUSTED && subCheck.result === YotiSessionDocument.SUBCHECK_PASS,
+  		);
+	
+  		return validChip;
+  	}
+	
+  	return false;
+  }
+	
+
+	/**
+	 The following documents will be scored a 3:
+	 	Non chipped Passports or passports with chips which are un-opened
+		UK or EU Drivers Licence
+		EEA National Identity Card (Without Chip being read)
+		
+	 The following documents will be scored a 4:
+	 	UK Passports or Passports where the chip has been opened
+		BRP
+		EEA National Identity Card (With Chip which has been read)
+	*/
+  private calculateStrengthScore(documentType: string, issuingCountry: string, documentContainsValidChip: boolean): number {
+  	if (issuingCountry === "GBR") {
+			switch (documentType) {
+				case "PASSPORT":
+					return documentContainsValidChip ? 4 : 3;
+				case "DRIVING_LICENCE":
+						return 3;
+				default:
+					throw new AppError(HttpCodesEnum.SERVER_ERROR, "Invalid documentType provided for issuingCountry", {
+						documentType, issuingCountry,
+					});
+			}
+		}
+		switch (documentType) {
+			case "PASSPORT":
+  			return 3;
+			case "DRIVING_LICENCE":
+  			return 3;
+			case "NATIONAL_ID":
+				return documentContainsValidChip ? 4 : 3;
+			case "RESIDENCE_PERMIT":
+				return 4;
+  		default:
+  			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Invalid documentType provided", {
+  				documentType, issuingCountry,
+  			});
+  	}
+  }
+	
+
+	/**
+		If the ID_DOCUMENT_AUTHENTICITY object has a recommendation value of approve, the document should be given a validity score of 2
+		If the ID_DOCUMENT_AUTHENTICITY has an recommendation value of approve and the object also includes an NFC check, the validity score will be 3. 
```

Comment:
let's add an explicit link to the JIRA or Confluence page where this requirement is defined?